core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -858,7 +858,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     // Use setInputPaths so that wholeTextFiles aligns with hadoopFile/textFile in taking
     // comma separated files as input. (see SPARK-7155)
     NewFileInputFormat.setInputPaths(job, path)
-    val updateConf = job.getConfiguration
+    val updateConf = SparkHadoopUtil.get.getConfigurationFromJobContext(job)
```

Comment:
@JoshRosen I didn't change `job.getConfiguration` last time because they uses `org.apache.hadoop.mapreduce.Job` whose `getConfiguration` is compatible. However, +1 for this change since it is good for maintenance.